components/browser/awesomebar/src/main/java/mozilla/components/browser/awesomebar/layout/DefaultSuggestionViewHolder.kt

```diff
@@ -58,6 +64,12 @@ internal sealed class DefaultSuggestionViewHolder {
                 suggestion.onSuggestionClicked?.invoke()
                 selectionListener.invoke()
             }
+
+            val hasTitle = !title.isNullOrBlank()
+            editView.visibility = if (hasTitle) View.VISIBLE else View.GONE
+            editView.setOnClickListener {
+                awesomeBar.editSuggestionListener?.invoke("$title ")
```

Comment:
There is a problem here when you add the space at the end through the view holder.

You can see from the gif below, pressing the "append button" multiple times adds a new space at the end of `EditToolbar`.

![image](https://user-images.githubusercontent.com/1370580/88352022-4a773880-cd26-11ea-8f5c-3acfeb9cfce7.png)

This is happening because the interactions with the `SearchActionProvider` that commits the `text` as a whole when clicked. We probably need to re-think how we want to add this listener on the Fenix side (see above comment as well).

Reply:
I've removed the space at the end for now. Should we maybe handle this in a separate PR?

Reply:
We could in the interest of landing this PR quicker!

Reply:
My thoughts, exactly.

Reply:
Marking as resolved, then.